frontend/src/lib/v2/WorkflowUtils.ts

```diff
@@ -78,6 +78,11 @@ export function convertYamlToV2PipelineSpec(template: string): PipelineSpec {
   // return pipelineSpec;
 }
 
+export function convertYamlToPlatformSpec(template: string) {
+  const platformSpecYaml = getPlatformDefFromYaml(template);
+  return platformSpecYaml ? PlatformSpec.fromJSON(platformSpecYaml) : undefined;
```

Comment:
I think we should use:

`return PlatformSpec.fromJSON(platformSpecYaml) ?? undefined;``

here.

Reply:
The yaml file without `platform_spec` will lead **platfromSpecYaml** to be `undefined`, which will throw an error when `PlatformSpec.fromJSON(platformSpecYaml)` is called.